harlow/sampling/fuzzy_lolavoronoi.py

```diff
@@ -133,12 +132,14 @@ def sample(
         # fit the surrogate model
         start_time = time.time()
 
-        self.surrogate_model.fit(points_x, points_y)
+        self.surrogate_models[0].fit(points_x, points_y)
         logger.info(
             f"Fitted the first surrogate model in {time.time() - start_time} sec."
         )
 
-        predicted_y = self.surrogate_model.predict(self.test_points_x, as_array=True)
+        predicted_y = self.surrogate_models[0].predict(
```

Comment:
Why is only the surrogate at index 0 used? (not only here but also in the rest of the code and in other samplers as well). 
I'd expect the following kind of behaviour:
- a number of surrogates is created for provided input-output combinations. This can be 1 ... NxM
- the sampling loop takes care of these different surrogates, which means that sampler should do it's sampling for each surrogate
- the sampling loop takes care administering the results for each surrogate-sampling result.